src/main/php/PHPMD/Renderer/AnsiRenderer.php

```diff
@@ -38,6 +38,7 @@ private function writeViolationsReport(Report $report)
         $previousFile = null;
         foreach ($report->getRuleViolations() as $violation) {
             if ($violation->getFileName() !== $previousFile) {
+                $this->writer->write('\n');
```

Comment:
We don't want an extra empty line for the first one:
```suggestion
                if ($previousFile !== null) {
                    $this->writer->write('\n');
                }
```